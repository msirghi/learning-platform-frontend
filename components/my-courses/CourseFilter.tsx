import React from 'react';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';
import FilterListIcon from '@material-ui/icons/FilterList';
import styles from '../../styles/modules/MyCourses.module.scss';

type Props = {
  onSearchInputChange: (val: string) => void;
  searchValue: string;
};

export const CourseFilter: React.FC<Props> = ({ searchValue, onSearchInputChange }) => {
  return (
    <div className={styles.myCoursesHeaderRightWrapper}>
      <TextField
        size='small'
        label='Search'
        onChange={(e) => onSearchInputChange(e.target.value)}
        value={searchValue}
      />
      <IconButton>
        <FilterListIcon />
      </IconButton>
    </div>
  );
};
