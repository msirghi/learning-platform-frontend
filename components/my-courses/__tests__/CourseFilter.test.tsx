import { TextField } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CourseFilter } from '../CourseFilter';

describe('CourseFilter component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<CourseFilter searchValue={'val'} onSearchInputChange={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire onSearchInputChange on input value change', () => {
    const spy = jest.fn();
    const wrapper = shallow(<CourseFilter searchValue={'val'} onSearchInputChange={spy} />);

    const input = wrapper.find(TextField);
    expect(input).toHaveLength(1);
    input
      .props()
      .onChange({ target: { value: 'value' } } as React.ChangeEvent<
        HTMLTextAreaElement | HTMLInputElement
      >);
    expect(spy).toBeCalled();
  });

  it('should have relevant initial input value', () => {
    const wrapper = shallow(
      <CourseFilter searchValue={'initial'} onSearchInputChange={jest.fn()} />
    );

    const input = wrapper.find(TextField);
    expect(input).toHaveLength(1);
    expect(input.props().value).toEqual('initial');
  });
});
