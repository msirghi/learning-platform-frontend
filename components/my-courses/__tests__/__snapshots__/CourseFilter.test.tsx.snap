// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseFilter component should match the snapshot 1`] = `
<div
  className="myCoursesHeaderRightWrapper"
>
  <WithStyles(ForwardRef(TextField))
    label="Search"
    onChange={[Function]}
    size="small"
    value="val"
  />
  <WithStyles(ForwardRef(IconButton))>
    <Memo([object Object]) />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
